### ISSUE TITLE
After setting new options list and refresh, showing "Nothing selected" 

### ISSUE THREAD

**Issue by insad** [snapappointments/bootstrap-select#2440]
After setting a new list of options on the original <select> (with one of the options with property "selected"), and calling <select>.selectpicker('refresh'), the message "Nothing selected" is shown, though the value of the element is correct.

Workaround:

```
<select>.empty();
.... fill with <option>, one of them <option .... selected="selected">
<select>.selectpicker('refresh');
const val = <select>.selectpicker('val');
<select>.selectpicker('val', val); // Forces the element to show the correct item
```

**Comment by caseyjhol** [comment]
Bug reports must include a live demo of the problem. Per our [contributing guidelines](https://github.com/snapappointments/bootstrap-select/blob/master/CONTRIBUTING.md), please create a reduced test case (you can use our preconfigured [Plunker](https://developer.snapappointments.com/bootstrap-select/playground/)), and report back with:

- your link
- Bootstrap version
- bootstrap-select version
- specific browser and OS details.

If you're not using the latest version of bootstrap-select, please try that as well.

**Comment by lcsqlpete** [comment]
I am having this issue too.

Attached is a file that illustrates the problem.  The select initially has no options. Click the "Click to populate select" button to load options.  I set the data-show-tick option to true so you can see that there is a selected option by clicking the select, but "Nothing selected" still shows.

Apparently github doesn't support .html files so the file is attached as a.txt file.

In addition, the docs describe the noneSelectedText option as "The text that is displayed when a multiple select has no selected options." but I see the text for selects that are not multiple selects.  Is this a bug or a documentation issue?

[NothingSelectedBug.txt](https://github.com/snapappointments/bootstrap-select/files/4554541/NothingSelectedBug.txt)

.

**Comment by caseyjhol** [comment]
This has already been fixed in v1.13.16: #2445. This is the main reason I ask for the version of bootstrap-select you're using, as well as the reason I ask you to try the latest version first.

**Comment by bhooby** [comment]
After spending many hours using single quotes for dynamic value fixed my issues.

before
$('#categoryParentID').selectpicker('val' , parentID);

After
$('#categoryParentID').selectpicker('val',''+parentID+'');
